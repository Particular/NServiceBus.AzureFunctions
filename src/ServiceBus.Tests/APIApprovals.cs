namespace AzureServiceBus.Tests
{
    using NServiceBus.AzureFunctions.ServiceBus;
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(ServiceBusTriggeredEndpointConfiguration).Assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
            Approver.Verify(publicApi);
        }
    }
}