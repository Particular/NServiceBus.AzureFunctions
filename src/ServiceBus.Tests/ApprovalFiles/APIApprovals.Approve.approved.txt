namespace NServiceBus.AzureFunctions.ServiceBus
{
    public class FunctionEndpoint : NServiceBus.Serverless.ServerlessEndpoint<Microsoft.Azure.WebJobs.ExecutionContext, NServiceBus.AzureFunctions.ServiceBus.ServiceBusTriggeredEndpointConfiguration>
    {
        public FunctionEndpoint(System.Func<Microsoft.Azure.WebJobs.ExecutionContext, NServiceBus.AzureFunctions.ServiceBus.ServiceBusTriggeredEndpointConfiguration> configurationFactory) { }
        public System.Threading.Tasks.Task Process(Microsoft.Azure.ServiceBus.Message message, Microsoft.Azure.WebJobs.ExecutionContext executionContext) { }
    }
    public class ServiceBusTriggeredEndpointConfiguration : NServiceBus.Serverless.ServerlessEndpointConfiguration
    {
        public ServiceBusTriggeredEndpointConfiguration(string connectionStringName, Microsoft.Azure.WebJobs.ExecutionContext executionContext) { }
        public NServiceBus.TransportExtensions<NServiceBus.AzureServiceBusTransport> Transport { get; }
    }
}